packages/f-checkout/src/components/Selector.vue

```diff
@@ -48,85 +27,34 @@ export default {
             return this.serviceType === CHECKOUT_METHOD_DELIVERY
                 ? this.$t('labels.deliveryOrderMethod')
                 : this.$t('labels.collectionOrderMethod');
-        }
-    },
-
-    watch: {
-        'fulfillment.times' (newValue) {
-            const selected = newValue.find(t => t.selected);
-            if (selected) {
-                this.selectedTime = selected.from;
-            }
+        },
+
+        /*
+        * Create an array from fulfillment times labels to
+        * display as options in dropdown
+        */
+        fulfillmentTimes () {
+            return this.fulfillment.times.map(time => time.label.text);
         }
     },
 
     methods: {
-        selectionChanged () {
-            this.fulfillment.times.forEach(el => { el.selected = false; });
-            const newSelected = this.fulfillment.times.find(t => t.from === this.selectedTime);
-            if (newSelected) {
-                newSelected.selected = true;
-            }
+        /*
+        * Update all fulfillment.times.selected to false
+        * Update chosen fulfillment.times.selected to true
+        */
+        selectionChanged (selectedTime) {
+            this.fulfillment.times.forEach(fulfillmentTime => {
+                fulfillmentTime.selected = fulfillmentTime.label.text === selectedTime;
+            });
         }
     }
 };
 </script>
 
 <style lang="scss" module>
-$form-label-colour                        : $grey--dark;
-$form-input-colour                        : $color-text;
-$form-input-bg                            : $white;
-$form-input-borderRadius                  : 3px;
-$form-input-borderWidth                   : 1px;
-$form-input-borderColour                  : $grey--light;
-$form-input-borderColour--focus           : $grey--dark;
-
-.o-form-select {
-    position: relative;
-    height: 60px;
-    margin: spacing(x2) 0;
-    padding: 0;
-    @include font-size(body-l);
-    font-family: $font-family-base;
-    color: $form-input-colour;
-    font-weight: $font-weight-base;
-    background-color: $form-input-bg;
-    border: $form-input-borderWidth solid $form-input-borderColour;
-    border-radius: $form-input-borderRadius;
-
-    .o-form-select-label {
-        display: block;
-        position: absolute;
-        top: 50%;
-        transform: translateY(-50%);
-        padding: spacing();
-        color: $form-label-colour;
-        cursor: pointer;
-    }
-
-    .o-form-select-input {
-        height: 100%;
-        width: 100%;
-        padding: spacing(x0.5);
-        border: none;
-        cursor: pointer;
-        color: $color-text;
-    }
-}
-
-/**
- * Modifier – .o-form-select--float
- *
- * Moves the select label to sit above chosen option
- */
-.o-form-select--float {
-    .o-form-select-label {
-        @include font-size(body-s);
-        top: 15px;
-    }
-
-    .o-form-select-input {
-        padding-top: spacing(x3);
-    }
+.c-selector {
```

Comment:
I would make this name more specific to being the checkout selector, as the styling is specific to that. So calling it `c-checkout-selector` (I'm assuming there's a `c-checkout` class in the parent component)